lib/iris/tests/test_netcdf.py

```diff
@@ -910,6 +910,65 @@ def test_std_name_lookup_fail(self):
         self.assertFalse('phenomenon_time' in iris.std_names.STD_NAMES)
 
 
+class TestCMIP6VolcelloLoad(tests.IrisTest):
+    def setUp(self):
+        dataset = nc.Dataset('issue_3367_volcello_test_file.nc', 'w')
+
+        dataset.createDimension('lat', 4)
+        dataset.createDimension('lon', 5)
+        dataset.createDimension('lev', 3)
+
+        latitudes = dataset.createVariable('lat', np.float64, ('lat',))
+        longitudes = dataset.createVariable('lon', np.float64, ('lon',))
+        levels = dataset.createVariable('lev', np.float64, ('lev',))
+        volcello = dataset.createVariable('volcello', np.float32,
+                                          ('lat', 'lon', 'lev'))
+
+        latitudes.standard_name = 'latitude'
+        latitudes.units = 'degrees_north'
+        latitudes.axis = 'Y'
+        latitudes[:] = np.linspace(-90, 90, 4)
+
+        longitudes.standard_name = 'longitude'
+        longitudes.units = 'degrees_east'
+        longitudes.axis = 'X'
+        longitudes[:] = np.linspace(0, 360, 5)
+
+        levels.standard_name = 'olevel'
+        levels.units = 'centimeters'
+        levels.positive = 'down'
+        levels.axis = 'Z'
+        levels[:] = np.linspace(0, 10**5, 3)
+
+        volcello.id = 'volcello'
+        volcello.out_name = 'volcello'
+        volcello.standard_name = 'ocean_volume'
+        volcello.units = 'm3'
+        volcello.realm = 'ocean'
+        volcello.frequency = 'fx'
+        volcello.cell_measures = 'area: areacello volume: volcello'
+        volcello = np.arange(4*5*3).reshape((4, 5, 3))
+
+        dataset.close()
+
+    def test_cmip6_volcello_load_issue_3367(self):
+        # a warning should be raised about missing areacello
+        areacello_str = "areacello" if six.PY3 else u"areacello"
+        volcello_str = "volcello" if six.PY3 else u"volcello"
+        expected_msg = "Missing CF-netCDF measure variable %r, " \
+                       "referenced by netCDF variable %r" \
+                       % (areacello_str, volcello_str)
+
+        with mock.patch('warnings.warn') as warn:
+            cube = iris.load_cube('issue_3367_volcello_test_file.nc')
```

Comment:
It's at least worth a comment here what the basic point of the test is
-- because the basic point is that the code did not fail, which is never that obvious !

Especially as the warning you're handling here is not the main point : I'm not even sure it's worth doing that, you could just let it happen, or maybe just mask it with warnings.catch_warnings / warnings.filterwarnings.